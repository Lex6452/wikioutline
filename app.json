{
    "name": "Outline",
    "description": "Open source wiki and knowledge base for growing teams",
    "website": "https://www.getoutline.com/",
    "repository": "https://github.com/outline/outline",
    "keywords": [
      "wiki",
      "team",
      "node",
      "markdown",
      "slack"
    ],
    "success_url": "/",
    "formation": {
      "web": {
        "quantity": 1,
        "size": "Hobby"
      }
    },
    "image": "heroku/node",
    "addons": [
      {
        "plan": "heroku-redis"
      },
      {
        "plan": "heroku-postgresql"
      }
    ],
    "scripts": {
      "postdeploy": "yarn sequelize db:migrate"
    },
    "env": {
      "SECRET_KEY": {
        "description": "A secret key",
        "generator": "secret"
      },
      "DEPLOYMENT": {
        "value": "self",
        "required": true
      },
      "ENABLE_UPDATES": {
        "value": true,
        "required": true
      },
      "SUBDOMAINS_ENABLED": {
        "value": false,
        "required": true,
        "description": "Running different subdomains. Likely not needed in single org mode"
      },
      "URL": {
        "description": "https://{your app name}.herokuapp.com",
        "required": false
      },
      "GOOGLE_CLIENT_ID": {
        "description": "See https://developers.google.com/identity/protocols/OAuth2 to create a new Google OAuth client. You must configure at least one of Slack or Google to control login.",
        "required": false
      },
      "GOOGLE_CLIENT_SECRET": {
        "description": "",
        "required": false
      },
      "GOOGLE_ALLOWED_DOMAINS": {
        "description": "Comma separated list of domains to be allowed (optional). If not set, all Google apps domains are allowed by default",
        "required": false
      },
      "SLACK_KEY": {
        "description": "See https://api.slack.com/apps to create a new Slack app. You must configure at least one of Slack or Google to control login.",
        "required": false
      },
      "SLACK_SECRET": {
        "description": "Your Slack client secret - d2dc414f9953226bad0a356cXXXXYYYY",
        "required": false
      },
      "SLACK_VERIFICATION_TOKEN": {
        "description": "Your Slack verification token - PLxk6OlXXXXXVj3YYYY",
        "required": false
      },
      "SLACK_APP_ID": {
        "description": "A0XXXXXXXXX",
        "required": false
      },
      "AWS_ACCESS_KEY_ID": {
        "description": "",
        "required": false
      },
      "AWS_SECRET_ACCESS_KEY": {
        "description": "",
        "required": false
      },
      "AWS_S3_UPLOAD_BUCKET_NAME": {
        "description": "yourbucket.example.com",
        "required": false
      },
      "AWS_S3_UPLOAD_BUCKET_URL": {
        "description": "Live web link to your bucket. For CNAMEs, https://yourbucket.example.com",
        "required": false
      },
      "AWS_S3_UPLOAD_MAX_SIZE": {
        "description": "",
        "required": false,
        "value": "26214400"
      },
      "SMTP_HOST": {
        "description": "smtp.example.com (optional)",
        "required": false
      },
      "SMTP_PORT": {
        "description": "1234 (optional)",
        "required": false
      },
      "SMTP_USERNAME": {
        "description": "me@example.com (optional)",
        "required": false
      },
      "SMTP_PASSWORD": {
        "description": "(optional)",
        "required": false
      },
      "SMTP_FROM_EMAIL": {
        "description": "wiki@example.com (optional)",
        "required": false
      },
      "SMTP_REPLY_EMAIL": {
        "description": "wikireply@example.com (optional)",
        "required": false
      },
      "GOOGLE_ANALYTICS_ID": {
        "required": false,
        "description": "UA-xxxx (optional)"
      },
      "BUGSNAG_KEY": {
        "required": false,
        "description": "(optional)"
      },
      "GITHUB_ACCESS_TOKEN": {
        "required": false,
        "description": "(optional)"
      }

    }
  }