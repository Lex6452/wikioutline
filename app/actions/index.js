// @flow
import type { TFunction } from "i18next";
import * as React from "react";
import type { Action } from "types";

export function actionToMenuItem(
  action: Action,
  context: { t: TFunction, event?: Event }
): ?Object {
  const { t } = context;

  function resolve<T>(value: any): T {
    if (typeof value === "function") {
      return value(context);
    }

    return value;
  }

  const resolvedIcon = resolve<React.Element<any>>(action.icon);
  const resolvedChildren = resolve<Action[]>(action.children);

  const item = {
    title: action.name({ t }),
    icon:
      resolvedIcon && action.iconInContextMenu !== false
        ? React.cloneElement(resolvedIcon, { color: "currentColor" })
        : undefined,
    onClick: action.perform
      ? () => action.perform && action.perform(context)
      : undefined,
    items: resolvedChildren
      ? resolvedChildren
          .map((a) => actionToMenuItem(a, context))
          .filter((a) => !!a)
      : undefined,
    visible: action.visible ? action.visible(context) : true,
    selected: action.selected ? action.selected(context) : undefined,
    // TODO
    // disabled
    // visible
    // selected
  };

  console.log(item);
  return item;
}

export function actionToKBar(action: Action, { t }: { t: TFunction }) {
  //
}
