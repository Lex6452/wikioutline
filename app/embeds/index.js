// @flow
import Airtable from './Airtable';
import Codepen from './Codepen';
import Figma from './Figma';
import Framer from './Framer';
import Gist from './Gist';
import InVision from './InVision';
import Loom from './Loom';
import Lucidchart from './Lucidchart';
import Marvel from './Marvel';
import ModeAnalytics from './ModeAnalytics';
import Numeracy from './Numeracy';
import RealtimeBoard from './RealtimeBoard';
import Spotify from './Spotify';
import Trello from './Trello';
import Typeform from './Typeform';
import Vimeo from './Vimeo';
import YouTube from './YouTube';

export default {
  Airtable,
  Codepen,
  Figma,
  Framer,
  Gist,
  InVision,
  Loom,
  Lucidchart,
  Marvel,
  ModeAnalytics,
  Numeracy,
  RealtimeBoard,
  Spotify,
  Trello,
  Typeform,
  Vimeo,
  YouTube,
};
