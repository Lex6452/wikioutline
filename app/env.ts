// @ts-expect-error ts-migrate(2339) FIXME: Property 'env' does not exist on type 'Window & ty... Remove this comment to see the full error message
const env = window.env;

export default env;
