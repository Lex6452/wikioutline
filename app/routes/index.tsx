import * as React from "react";
import { Switch } from "react-router-dom";
import DelayedMount from "components/DelayedMount";
import FullscreenLoading from "components/FullscreenLoading";
// @ts-expect-error ts-migrate(7034) FIXME: Variable 'Route' implicitly has type 'any' in some... Remove this comment to see the full error message
import Route from "components/ProfiledRoute";
// @ts-expect-error ts-migrate(2307) FIXME: Cannot find module 'utils/routeHelpers' or its cor... Remove this comment to see the full error message
import { matchDocumentSlug as slug } from "utils/routeHelpers";

const Authenticated = React.lazy(
  () =>
    import(
      /* webpackChunkName: "authenticated" */
      "components/Authenticated"
    )
);
const AuthenticatedRoutes = React.lazy(
  () =>
    import(
      /* webpackChunkName: "authenticated-routes" */
      "./authenticated"
    )
);
const SharedDocument = React.lazy(
  () =>
    import(
      /* webpackChunkName: "shared-document" */
      "scenes/Document/Shared"
    )
);
const Login = React.lazy(
  () =>
    import(
      /* webpackChunkName: "login" */
      "scenes/Login"
    )
);
const Logout = React.lazy(
  () =>
    import(
      /* webpackChunkName: "logout" */
      "scenes/Logout"
    )
);

export default function Routes() {
  return (
    <React.Suspense
      fallback={
        <DelayedMount delay={2000}>
          <FullscreenLoading />
        </DelayedMount>
      }
    >
      <Switch>
        // @ts-expect-error ts-migrate(7005) FIXME: Variable 'Route' implicitly
        has an 'any' type.
        <Route exact path="/" component={Login} />
        // @ts-expect-error ts-migrate(7005) FIXME: Variable 'Route' implicitly
        has an 'any' type.
        <Route exact path="/create" component={Login} />
        // @ts-expect-error ts-migrate(7005) FIXME: Variable 'Route' implicitly
        has an 'any' type.
        <Route exact path="/logout" component={Logout} />
        // @ts-expect-error ts-migrate(7005) FIXME: Variable 'Route' implicitly
        has an 'any' type.
        <Route exact path="/share/:shareId" component={SharedDocument} />
        // @ts-expect-error ts-migrate(7005) FIXME: Variable 'Route' implicitly
        has an 'any' type.
        <Route
          exact
          path={`/share/:shareId/doc/${slug}`}
          component={SharedDocument}
        />
        <Authenticated>
          <AuthenticatedRoutes />
        </Authenticated>
      </Switch>
    </React.Suspense>
  );
}
