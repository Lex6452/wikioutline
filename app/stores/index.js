// @flow
import RootStore from 'stores/RootStore';

const stores = new RootStore();
console.log(stores);

export default stores;
