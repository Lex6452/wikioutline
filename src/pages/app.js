import React from "react";
import OutlineIndex from 'outline/index';

export default function App() {
	const isSsr = typeof window === 'undefined';

	return (
		<div id="root">
			{isSsr ? (
				<></>
			) : (
				<OutlineIndex/>
			)}
		</div>
	);
}
