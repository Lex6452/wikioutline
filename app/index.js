// @flow
import * as React from 'react';
import { render } from 'react-dom';
import { Provider } from 'mobx-react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from 'react-router-dom';

import stores from 'stores';
import globalStyles from 'shared/styles/globals';
import 'shared/styles/prism.css';

import Home from 'scenes/Home';
import Dashboard from 'scenes/Dashboard';
import Starred from 'scenes/Starred';
import Drafts from 'scenes/Drafts';
import Collection from 'scenes/Collection';
import Document from 'scenes/Document';
import Search from 'scenes/Search';
import Settings from 'scenes/Settings';
import Users from 'scenes/Settings/Users';
import Billing from 'scenes/Settings/Billing';
import Slack from 'scenes/Settings/Slack';
import Tokens from 'scenes/Settings/Tokens';
import SlackAuth from 'scenes/SlackAuth';
import ErrorAuth from 'scenes/ErrorAuth';
import Error404 from 'scenes/Error404';

import ErrorBoundary from 'components/ErrorBoundary';
import ScrollToTop from 'components/ScrollToTop';
import Layout from 'components/Layout';
import Auth from 'components/Auth';
import RouteSidebarHidden from 'components/RouteSidebarHidden';

import { matchDocumentSlug } from 'utils/routeHelpers';
import { BILLING_ENABLED } from 'shared/environment';

let DevTools;
if (__DEV__) {
  DevTools = require('mobx-react-devtools').default; // eslint-disable-line global-require
}

const notFoundSearch = () => <Search notFound />;
const DocumentNew = () => <Document newDocument />;
const RedirectDocument = ({ match }: { match: Object }) => (
  <Redirect to={`/doc/${match.params.documentSlug}`} />
);

globalStyles();

render(
  <React.Fragment>
    <ErrorBoundary>
      <Provider {...stores}>
        <Router>
          <ScrollToTop>
            <Switch>
              <Route exact path="/" component={Home} />
              <Route exact path="/auth/slack" component={SlackAuth} />
              <Route exact path="/auth/slack/commands" component={SlackAuth} />
              <Route exact path="/auth/error" component={ErrorAuth} />

              <Auth>
                <Layout>
                  <Switch>
                    <Route exact path="/dashboard" component={Dashboard} />
                    <Route exact path="/starred" component={Starred} />
                    <Route exact path="/drafts" component={Drafts} />
                    <Route exact path="/settings" component={Settings} />
                    <Route exact path="/settings/users" component={Users} />
                    {BILLING_ENABLED && (
                      <Route
                        exact
                        path="/settings/billing"
                        component={Billing}
                      />
                    )}
                    <Route exact path="/settings/tokens" component={Tokens} />
                    <Route
                      exact
                      path="/settings/integrations/slack"
                      component={Slack}
                    />

                    <Route
                      exact
                      path="/collections/:id"
                      component={Collection}
                    />
                    <Route
                      exact
                      path={`/d/${matchDocumentSlug}`}
                      component={RedirectDocument}
                    />
                    <Route
                      exact
                      path={`/doc/${matchDocumentSlug}`}
                      component={Document}
                    />
                    <Route
                      exact
                      path={`/doc/${matchDocumentSlug}/move`}
                      component={Document}
                    />

                    <Route exact path="/search" component={Search} />
                    <Route exact path="/search/:query" component={Search} />

                    <Route path="/404" component={Error404} />

                    <RouteSidebarHidden
                      exact
                      path={`/doc/${matchDocumentSlug}/edit`}
                      component={Document}
                    />
                    <RouteSidebarHidden
                      exact
                      path="/collections/:id/new"
                      component={DocumentNew}
                    />
                    <Route component={notFoundSearch} />
                  </Switch>
                </Layout>
              </Auth>
            </Switch>
          </ScrollToTop>
        </Router>
      </Provider>
    </ErrorBoundary>
    {DevTools && <DevTools position={{ bottom: 0, right: 0 }} />}
  </React.Fragment>,
  document.getElementById('root')
);

window.addEventListener('load', async () => {
  // installation does not use Google Analytics, or tracking is blocked on client
  // no point loading the rest of the analytics bundles
  if (!process.env.GOOGLE_ANALYTICS_ID || !window.ga) return;

  // https://github.com/googleanalytics/autotrack/issues/137#issuecomment-305890099
  await import('autotrack/autotrack.js');

  window.ga('require', 'outboundLinkTracker');
  window.ga('require', 'urlChangeTracker');
});
