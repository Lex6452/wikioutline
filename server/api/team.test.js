/* eslint-disable flowtype/require-valid-file-annotation */
import TestServer from 'fetch-test-server';

import app from '..';

import { flushdb, seed } from '../test/support';

const server = new TestServer(app.callback());

beforeEach(flushdb);
afterAll(server.close);

describe('#team.users', async () => {
  it('should return teams paginated user list', async () => {
    const { admin } = await seed();

    const res = await server.post('/api/team.users', {
      body: { token: admin.getJwtToken() },
    });
    const body = await res.json();

    expect(res.status).toEqual(200);
    expect(body).toMatchSnapshot();
  });

  it('should require admin for detailed info', async () => {
    const { user } = await seed();
    const res = await server.post('/api/team.users', {
      body: { token: user.getJwtToken() },
    });
    const body = await res.json();

    expect(res.status).toEqual(200);
    expect(body).toMatchSnapshot();
  });
});

describe('#team.addAdmin', async () => {
  it('should promote a new admin', async () => {
    const { admin, user } = await seed();

    const res = await server.post('/api/team.addAdmin', {
      body: { token: admin.getJwtToken(), user: user.id },
    });
    const body = await res.json();

    expect(res.status).toEqual(200);
    expect(body).toMatchSnapshot();
  });

  it('should require admin', async () => {
    const { user } = await seed();
    const res = await server.post('/api/team.addAdmin', {
      body: { token: user.getJwtToken(), user: user.id },
    });
    const body = await res.json();

    expect(res.status).toEqual(403);
    expect(body).toMatchSnapshot();
  });
});

describe('#team.removeAdmin', async () => {
  it('should demote an admin', async () => {
    const { admin, user } = await seed();
    await user.update({ isAdmin: true }); // Make another admin

    const res = await server.post('/api/team.removeAdmin', {
      body: {
        token: admin.getJwtToken(),
        user: user.id,
      },
    });
    const body = await res.json();

    expect(res.status).toEqual(200);
    expect(body).toMatchSnapshot();
  });

  it("shouldn't demote admins if only one available ", async () => {
    const { admin } = await seed();

    const res = await server.post('/api/team.removeAdmin', {
      body: {
        token: admin.getJwtToken(),
        user: admin.id,
      },
    });
    const body = await res.json();

    expect(res.status).toEqual(400);
    expect(body).toMatchSnapshot();
  });

  it('should require admin', async () => {
    const { user } = await seed();
    const res = await server.post('/api/team.addAdmin', {
      body: { token: user.getJwtToken(), user: user.id },
    });
    const body = await res.json();

    expect(res.status).toEqual(403);
    expect(body).toMatchSnapshot();
  });
});
