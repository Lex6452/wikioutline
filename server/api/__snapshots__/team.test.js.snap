// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#team.addAdmin should promote a new admin 1`] = `
Object {
  "data": Object {
    "avatarUrl": "http://example.com/avatar.png",
    "email": "user1@example.com",
    "id": "46fde1d4-0050-428f-9f0b-0bf77f4bdf61",
    "isAdmin": true,
    "name": "User 1",
    "username": "user1",
  },
  "ok": true,
  "status": 200,
}
`;

exports[`#team.addAdmin should require admin 1`] = `
Object {
  "error": "admin_required",
  "message": "An admin role is required to access this resource",
  "ok": false,
  "status": 403,
}
`;

exports[`#team.removeAdmin should demote an admin 1`] = `
Object {
  "data": Object {
    "avatarUrl": "http://example.com/avatar.png",
    "email": "user1@example.com",
    "id": "46fde1d4-0050-428f-9f0b-0bf77f4bdf61",
    "isAdmin": false,
    "name": "User 1",
    "username": "user1",
  },
  "ok": true,
  "status": 200,
}
`;

exports[`#team.removeAdmin should require admin 1`] = `
Object {
  "error": "admin_required",
  "message": "An admin role is required to access this resource",
  "ok": false,
  "status": 403,
}
`;

exports[`#team.removeAdmin shouldn't demote admins if only one available  1`] = `
Object {
  "error": "validation_error",
  "message": "At least one admin is required",
  "ok": false,
  "status": 400,
}
`;

exports[`#team.users should require admin for detailed info 1`] = `
Object {
  "data": Array [
    Object {
      "avatarUrl": "http://example.com/avatar.png",
      "id": "fa952cff-fa64-4d42-a6ea-6955c9689046",
      "name": "Admin User",
      "username": "admin",
    },
    Object {
      "avatarUrl": "http://example.com/avatar.png",
      "id": "46fde1d4-0050-428f-9f0b-0bf77f4bdf61",
      "name": "User 1",
      "username": "user1",
    },
  ],
  "ok": true,
  "pagination": Object {
    "limit": 15,
    "nextPath": "/api/team.users?limit=15&offset=15",
    "offset": 0,
  },
  "status": 200,
}
`;

exports[`#team.users should return teams paginated user list 1`] = `
Object {
  "data": Array [
    Object {
      "avatarUrl": "http://example.com/avatar.png",
      "email": "admin@example.com",
      "id": "fa952cff-fa64-4d42-a6ea-6955c9689046",
      "isAdmin": true,
      "name": "Admin User",
      "username": "admin",
    },
    Object {
      "avatarUrl": "http://example.com/avatar.png",
      "email": "user1@example.com",
      "id": "46fde1d4-0050-428f-9f0b-0bf77f4bdf61",
      "isAdmin": false,
      "name": "User 1",
      "username": "user1",
    },
  ],
  "ok": true,
  "pagination": Object {
    "limit": 15,
    "nextPath": "/api/team.users?limit=15&offset=15",
    "offset": 0,
  },
  "status": 200,
}
`;
